### PR TITLE
vi mode: support increment/decrement of numbers

### PR DESCRIPTION
**Pull request by DevAtDawn**
## Description
use ctrl+x ctrl+a to increase/decrease a number

Fixes issue #8320

## TODOs:

- [ ] Changes to fish usage are reflected in user documentation/manpages.
- [ ] Tests have been added for regressions fixed
- [ ] User-visible changes noted in CHANGELOG.rst

### CODE REVIEW

**Comment by d-e-s-o** [comment]
For what it's worth, I had also gotten my own version implemented by now, just haven't gotten around to upstreaming it: https://github.com/fish-shell/fish-shell/compare/master...d-e-s-o:fish-shell:topic/increment-decrement

Haven't looked at the changes at hand to judge which one is nicer.

File: share/functions/increment_decrement.fish
Comment (ridiculousfish) [inline share/functions/increment_decrement.fish:1]: As this function is specific to these key bindings, let's name it something unwieldy like `__fish_inc_dec_number_under_cursor`.

File: share/functions/increment_decrement.fish [hunk 1]
```diff
@@ -0,0 +1,80 @@
+function increment_decrement
+    set current_buffer (commandline -b)
```
Comment (ridiculousfish) [inline share/functions/increment_decrement.fish:2]: Please add comments explaining what is happening, and how to call the function

**Review by ridiculousfish — CHANGES_REQUESTED** [review]
Thank you for tackling this. Please add a test if possible to the bind.py tests, [in the vi section](https://github.com/fish-shell/fish-shell/blob/a6c8cd6e85f5f7092b47e28e9963cb3268a2b5be/tests/pexpects/bind.py#L71)

**Pull request closed without merging**